### PR TITLE
asBools, asBool, and chained apply on asBools

### PR DESCRIPTION
**Pull request by jackkoenig**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: API modification

**Development Phase**: implementation

**Release Notes**
* toBools -> asBools (toBools deprecated)
* toBool -> asBool (toBool deprecated)
* Fix chaining of apply after `.asBools`

### CODE REVIEW

File: chiselFrontend/src/main/scala/chisel3/core/Bits.scala [hunk 1]
```diff
@@ -78,6 +78,15 @@ abstract class Element extends Data {
   */
 private[chisel3] sealed trait ToBoolable extends Element {
 
+  /** Casts this $coll to a [[Bool]]
```
Comment (edwardcwang) [inline chiselFrontend/src/main/scala/chisel3/core/Bits.scala]: Is `$coll` meant to be substituted?

Comment (jackkoenig) [reply chiselFrontend/src/main/scala/chisel3/core/Bits.scala]: @seldridge added this when he improved the ScalaDoc, I just copied it from `toBool`

File: chiselFrontend/src/main/scala/chisel3/core/Bits.scala [hunk 2]
```diff
@@ -412,13 +429,17 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends Element wi
     do_asUInt
   }
 
-  final def do_toBool(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = {
+  final def do_asBool(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Bool = {
     width match {
       case KnownWidth(1) => this(0)
-      case _ => throwException(s"can't covert UInt<$width> to Bool")
+      case _ => throwException(s"can't covert ${this.getClass.getSimpleName}$width to Bool")
```
[the remark this answers is not in the document]
Comment (jackkoenig) [reply chiselFrontend/src/main/scala/chisel3/core/Bits.scala]: Agreed, I think a way to render unbound types as well as bound values in a more useful way would be very valuable. The whole `UInt@4fd5e7a` is pretty worthless most of the time

[the remark this answers is not in the document]
Comment (ducky64) [reply chiselFrontend/src/main/scala/chisel3/core/Bits.scala]: Created a tracking issue for that in #953

**Comment by jackkoenig** [comment]
retest this please

**Pull request merged**